drivers/clk/clk-hifiberry-dacpro.c

```diff
@@ -69,6 +79,88 @@ static long clk_hifiberry_dacpro_round_rate(struct clk_hw *hw,
 	return actual_rate;
 }
 
+static int clk_hifiberry_dacpro_is_prepared(struct clk_hw *hw)
+{
+	struct clk_hifiberry_hw *clk = to_hifiberry_clk(hw);
+
+	return (clk->prepared) ? 1 : 0;
+}
+
+static int clk_hifiberry_dacpro_prepare(struct clk_hw *hw)
+{
+	struct clk_hifiberry_hw *clk = to_hifiberry_clk(hw);
+	int ret = 0;
+
+	/*
+	 * The gpios are handled here to avoid any dependencies on
+	 * probe. the gpios will be released in the .remove.
+	 *
+	 * The user of the clock should verify with the PCM512
+	 * registers that the clock are actually present and stable.
+	 * This driver only toggles the relevant GPIOs.
+	 */
+	if (!clk->gpio_initialized) {
+
+		clk->gpio_44 = gpiod_get(NULL, "PCM512x-GPIO6",
+					 GPIOD_OUT_LOW);
+		if (IS_ERR(clk->gpio_44)) {
+			pr_err("gpio44 not found\n");
+			return PTR_ERR(clk->gpio_44);
+		}
+
+		clk->gpio_48 = gpiod_get(NULL, "PCM512x-GPIO3",
+					 GPIOD_OUT_LOW);
+		if (IS_ERR(clk->gpio_48)) {
+			pr_err("gpio48 not found\n");
```

Comment:
not releasing `gpio_44` here?

Reply:
the gpios will be released in the .remove.

Reply:
don't think so, only if `clk->gpio_initialized == true` and it won't be the case here.

Reply:
ok, fixed